src/NLog/LogEventInfo.cs

```diff
@@ -523,6 +536,31 @@ private void CalcFormattedMessage()
             }
         }
 
+        private bool ResetFormattedProperties()
+        {
+            if (this.properties != null && this.properties.Count > 0 && this.logMessageFormatter.HasProperties(this))
```

Comment:
`logMessageFormatter.HasProperties` checks if the MessageFormatter can produce-properties.

It is an optimization for the case where structured-logging has been disabled (or the LogEventInfo doesn't match the profile).